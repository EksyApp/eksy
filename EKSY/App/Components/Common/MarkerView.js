import React, {Component} from 'react'
import {View, StyleSheet} from "react-native";
import Pointer from "./Pointer";
import Label from "./Label";
import TextArea from "./TextArea";
import PictureSwiper from "./PictureSwiper";
import Message from "./Message";
import PropTypes from 'prop-types';
import {MarkerShape, RegionShape} from "../../Utils/PropTypeShapes";

export default class MarkerView extends Component {
	
	getStatusMessage() {
		switch(this.props.marker.status) {
			case -1:
				return "Rejected"
			case 0:
				return "Waiting for confirmation"
			case 1:
				return "Accepted"
		}
	}
	
	render() {
		return(
				<View>
					<Pointer
							initialRegion={{...this.props.currentRegion, latitude: this.props.marker.latitude, longitude: this.props.marker.longitude}}
							style={styles.map}
					/>
					<Label>{this.props.marker.title}</Label>
					<Message>{this.getStatusMessage()}</Message>
					<TextArea>{this.props.marker.text}</TextArea>
					{this.props.marker.images && <PictureSwiper style={styles.swiper} data={this.props.marker.images}/>}
					{this.props.marker.filters && this.props.marker.filters.map((filter) => <Message>{filter}</Message>)}
				</View>
		)
	}
	
}

const styles = StyleSheet.create({
	map: {
		width: '100%',
		height: 300,
		marginBottom: 30
	},
	
	swiper: {
		width: '100%',
		height: 300,
		marginTop: 30
	}
})

MarkerView.propTypes = {
	marker: MarkerShape,
	currentRegion: RegionShape
}