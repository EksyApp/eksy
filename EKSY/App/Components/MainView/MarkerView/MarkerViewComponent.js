import React, {Component} from 'react'
import {Modal, ScrollView, TouchableWithoutFeedback, View, Dimensions, StyleSheet} from 'react-native'
import Label from "../../Common/Label";
import ViewMoreText from "../../Common/ViewMoreText";
import * as Theme from "../../../Theme/Colors";
import Divider from "../../Common/Divider";
import Picture from "../../Common/Picture";
import {Actions} from 'react-native-router-flux'
import PictureSwiper from "../../Common/PictureSwiper";
import {Icon} from 'react-native-elements'
import TextArea from "../../Common/TextArea";


const Screen = {
	width: Dimensions.get('window').width,
	height: Dimensions.get('window').height - 75
}

class MarkerViewComponent extends Component {
	
	
	constructor(props) {
		super(props)
		this.itemWidth = (0.85 * Screen.width * 2) / 3
		this.sliderWidth = 0.85 * Screen.width
		this.state = {
			isCollapsed: true
		}
	}
	
	_setCollapsed = () => {
		if (!this.state.isCollapsed) {
			this.setState({isCollapsed: true})
		}
	}
	
	_setExpanded = () => {
		if (this.state.isCollapsed) {
			this.setState({isCollapsed: false})
		}
	}
	
	
	_renderImages() {
		if (this.props.marker.images && this.props.marker.images.length > 0) {
			return (
					<View
							style={[!this.state.isCollapsed && styles.renderImagesCollapsed, this.state.isCollapsed && styles.renderImagesExpanded]}>
						<View style={styles.labelWrapper}>
							<Divider/>
						</View>
						
						<PictureSwiper data={this.props.marker.images}/>
					</View>
			)
		} else {
			return null
		}
	}
	
	_renderText() {
		if (this.props.marker.images && this.props.marker.images.length > 0) {
			return (<ViewMoreText
							numberOfLines={1}
							afterCollapse={this._setCollapsed}
							afterExpand={this._setExpanded}
					>
						{this.props.marker.text}
					</ViewMoreText>
			)
		} else {
			return (
					<TextArea>
								{this.props.marker.text}
					</TextArea>
			)
		}
		
	}
	
	_renderEditingIcon() {
		if (this.props.user && this.props.marker && this.props.marker.creationInfo && this.props.user.uid === this.props.marker.creationInfo.user) {
			return(
					<Icon
							name="edit"
							size={25}
							containerStyle={styles.editIcon}
							onPress={() => {
								Actions.editMarker()
								this.props.setMarkerViewHidden()
							}}
					/>
			)
		}
	}
	
	_renderTitleAndText() {
		return (
				<View
						style={this.state.isCollapsed ? styles.titleAndTextWrapperCollapsed : styles.titleAndTextWrapperExpanded}
				>
					<View style={styles.titleContainer}>
						<Label style={styles.title}>{this.props.marker.title}</Label>
						{this._renderEditingIcon()}
					</View>
					
					<ScrollView>
						{this._renderText()}
					</ScrollView>
				</View>
		)
	}
	
	render() {
		return (
				<TouchableWithoutFeedback onPress={this.props.setMarkerViewHidden}>
					<Modal
							visible={this.props.markerViewVisible}
							animationType={'fade'}
							transparent
							onRequestClose={this.props.setMarkerViewHidden}
					>
						<View style={styles.content}>
							{this._renderTitleAndText()}
							{this._renderImages()}
						</View>
					</Modal>
				</TouchableWithoutFeedback>
		)
	}
}


const styles = StyleSheet.create({
	
	content: {
		backgroundColor: Theme.backgroundColor,
		flex: 1,
		justifyContent: 'center',
		alignItems: 'center',
		borderRadius: 20,
		margin: 20
	},
	
	renderImagesCollapsed: {
		flex: 1,
		justifyContent: 'center'
	},
	renderImagesExpanded: {
		flex: 5,
		justifyContent: 'center',
		margin: 5
	},
	
	titleAndTextWrapperCollapsed: {
		flex: 1,
		alignItems: 'center',
		margin: 10
	},
	
	titleAndTextWrapperExpanded: {
		// flex: 0,
		alignItems: 'center',
		margin: 10,
	},
	
	swiper: {
		// width: 300,
		// margin: 20,
		width: '100%',
		height: '100%'
		// justifyContent: 'center',
		// alignItems: 'center',
	},
	
	title: {
		fontSize: 22,
		fontWeight: 'bold',
		color: Theme.accentColor
	},
	
	labelWrapper: {
		padding: 10
	},
	
	listStyle: {
		margin: 10
	},
	
	closeButton: {
		position: 'absolute',
		left: Screen.width - 120
	},
	
	editIcon: {
		position: 'absolute',
		top: 0,
		right: 0,
		
	},
	
	titleContainer: {
		width: '100%'
	}
})

export default MarkerViewComponent