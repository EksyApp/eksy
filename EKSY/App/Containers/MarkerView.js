import React, {Component} from 'react'
import {View, ScrollView, Text, StyleSheet} from 'react-native'
import Header from "../Components/Header";
import PictureList from '../Components/PictureList'
import * as ReactActions from '../Actions'
import {connect} from 'react-redux'
import TextArea from '../Components/TextArea'
import * as Theme from '../Theme'
import Divider from '../Components/Divider'
import Label from '../Components/Label'

export class MarkerView extends Component {

	_renderImages() {
		if(this.props.marker.images.length > 0) {
			return(
					<View>
						<Divider />
						<Label>Images</Label>
						<PictureList data={this.props.marker.images} listStyle={style.listStyle} imageContainerStyle={style.imageContainer} />
					</View>
			)
		} else {
			return null
		}
	}

	render() {
		return(
				<View style={style.container}>
					<Header backButton title={this.props.marker.title}/>
					<ScrollView>
						<View>
							<TextArea>
								{this.props.marker.text}
							</TextArea>
						</View>

						{this._renderImages()}

					</ScrollView>
				</View>
		)
	}

}

const style = StyleSheet.create({
	imageContainer: {
		width: '100%',
	},

	listStyle: {
		margin: 10
	},

	container: {
		backgroundColor: Theme.backgroundColor,
		flex: 1
	},

})

const mapStateToProps = (state) => {
	return {
		marker: state.map.selectedMarker.marker
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(MarkerView)
