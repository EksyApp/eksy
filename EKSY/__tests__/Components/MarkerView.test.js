import React from 'react';
import renderer from 'react-test-renderer';
import MarkerView from "../../App/Components/Common/MarkerView";

jest.mock('react-native-maps', () => {
	const React = require.requireActual('react');
	const MapView = require.requireActual('react-native-maps');
	
	class MockCallout extends React.Component {
		render() {
			return React.createElement('Callout', this.props, this.props.children);
		}
	}
	
	class MockMarker extends React.Component {
		render() {
			return React.createElement('Marker', this.props, this.props.children);
		}
	}
	
	class MockMapView extends React.Component {
		render() {
			return React.createElement('MapView', this.props, this.props.children);
		}
	}
	
	class MockMapViewAnimated extends React.Component {
		render() {
			return React.createElement('MapView', this.props, this.props.children);
		}
	}
	
	MockCallout.propTypes = MapView.Callout.propTypes;
	MockMarker.propTypes = MapView.Marker.propTypes;
	MockMapView.propTypes = MapView.propTypes;
	MockMapView.Marker = MockMarker;
	MockMapView.Callout = MockCallout;
	MockMapView.Animated = MockMapViewAnimated;
	return MockMapView;
});

test('MarkerView renders correctly', () => {
	const tree = renderer.create(
			<MarkerView
					marker={{
						filters: [],
						images : [],
						key: '' ,
						latitude: 1,
						longitude: 1,
						status: 1 ,
						text:  '' ,
						title:  ''
					}}
					currentRegion= {{latitude: 1, longitude: 1, latitudeDelta: 1, longitudeDelta: 1}}
			/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});