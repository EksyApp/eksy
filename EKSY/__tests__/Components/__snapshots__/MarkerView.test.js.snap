// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkerView renders correctly 1`] = `
<View>
  <MapView
    initialRegion={
      Object {
        "latitude": 1,
        "latitudeDelta": 1,
        "longitude": 1,
        "longitudeDelta": 1,
      }
    }
    style={
      Object {
        "height": 300,
        "marginBottom": 30,
        "width": "100%",
      }
    }
  >
    <Marker
      coordinate={
        Object {
          "latitude": 1,
          "latitudeDelta": 1,
          "longitude": 1,
          "longitudeDelta": 1,
        }
      }
    />
  </MapView>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "rgb(101,114,122)",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "rgb(89,85,78)",
          "fontSize": 18,
          "textAlign": "center",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    Accepted
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "rgb(89,85,78)",
          "fontSize": 18,
          "margin": 10,
        },
        undefined,
      ]
    }
  >
    
  </Text>
  <View
    onLayout={[Function]}
    style={
      Array [
        Array [
          Object {
            "height": "100%",
            "width": "100%",
          },
          Object {
            "height": 300,
            "marginTop": 30,
            "width": "100%",
          },
        ],
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      bounces={false}
      horizontal={true}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={true}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
    >
      <View />
    </RCTScrollView>
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "bottom": 0,
            "flexDirection": "row",
            "flexWrap": "wrap",
            "height": 32,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
